packages/toolpad-app/src/server/data.ts

```diff
@@ -531,16 +534,25 @@ export async function loadRenderTree(appId: string, version: VersionOrPreview =
 
 export async function duplicateApp(id: string, name: string): Promise<AppMeta> {
   const dom = await loadPreviewDom(id);
-  const duplicateCount = await prismaClient.app.count({
-    where: { name: { startsWith: `${name} (copy`, endsWith: ')' } },
-  });
-  const duplicateName =
-    duplicateCount === 0 ? `${name} (copy)` : `${name} (copy ${duplicateCount + 1})`;
-  const newApp = await createApp(duplicateName, {
+  const appFromDom: CreateAppOptions = {
     from: {
       kind: 'dom',
       dom,
     },
-  });
-  return newApp;
+  };
+  try {
+    const newApp = await createApp(name, appFromDom);
+    return newApp;
+  } catch (error) {
+    if (error instanceof Error && error.cause === TOOLPAD_UNIQUE_CONSTRAINT_VIOLATION) {
```

Comment:
`TOOLPAD_UNIQUE_CONSTRAINT_VIOLATION` is an implementation detail. If we're adding our own error codes, let's do `ERR_APP_EXISTS`